.github/workflows/pr-merge-notification.js

```diff
@@ -0,0 +1,40 @@
+const { slackNotification, getLocalConfigs } = require('./helpers.js');
+
+async function main({ github, context } = {}) {
+    if (!github || !context) {
+        throw new Error("GitHub context is missing. Ensure you are running in the correct environment.");
+    }
+
+    if (process.env.LOCAL_RUN) {
+        console.log("Local run detected. Loading local configurations...");
+        const localConfigs = getLocalConfigs();
+        github = localConfigs.github;
+        context = localConfigs.context;
+    }
+
+    const { pull_request } = context.payload;
+    if (!pull_request) {
+        console.log("No pull_request found in context payload. Skipping notification.");
+        return;
+    }
+
+    const { number, title, html_url, base } = pull_request;
+    const isStage = base.ref === 'stage';
+    const prefix = isStage
+        ? ':merged: PR merged to stage:'
+        : ':rocket: Production release:';
```

Comment:
Given that we already filter out all non-stage and non-prod PRs in the yaml that should be fine `(github.event.pull_request.base.ref == 'stage' || github.event.pull_request.base.ref == 'main')`